src/huggingface_hub/hf_api.py

```diff
@@ -972,56 +1004,103 @@ def model_info(
         d = r.json()
         return ModelInfo(**d)
 
-    def list_repo_files(
+    def dataset_info(
         self,
         repo_id: str,
         revision: Optional[str] = None,
-        repo_type: Optional[str] = None,
         token: Optional[str] = None,
         timeout: Optional[float] = None,
-    ) -> List[str]:
+    ) -> DatasetInfo:
         """
-        Get the list of files in a given repo.
+        Get info on one specific dataset on huggingface.co
+
+        Dataset can be private if you pass an acceptable token.
         """
-        if repo_type is None or repo_type == "model":
-            info = self.model_info(
-                repo_id, revision=revision, token=token, timeout=timeout
-            )
-        elif repo_type == "dataset":
-            info = self.dataset_info(
-                repo_id, revision=revision, token=token, timeout=timeout
-            )
-        else:
-            raise ValueError("Spaces are not available yet.")
+        if token is None:
+            token = HfFolder.get_token()
 
-        return [f.rfilename for f in info.siblings]
+        path = (
+            f"{self.endpoint}/api/datasets/{repo_id}"
+            if revision is None
+            else f"{self.endpoint}/api/datasets/{repo_id}/revision/{revision}"
+        )
+        headers = {"authorization": f"Bearer {token}"} if token is not None else None
+        r = requests.get(path, headers=headers, timeout=timeout)
+        r.raise_for_status()
+        d = r.json()
+        return DatasetInfo(**d)
 
-    def dataset_info(
+    def space_info(
         self,
         repo_id: str,
         revision: Optional[str] = None,
         token: Optional[str] = None,
         timeout: Optional[float] = None,
-    ) -> DatasetInfo:
+    ) -> SpaceInfo:
         """
-        Get info on one specific dataset on huggingface.co
+        Get info on one specific Space on huggingface.co
 
-        Dataset can be private if you pass an acceptable token.
+        Space can be private if you pass an acceptable token.
         """
         if token is None:
             token = HfFolder.get_token()
 
         path = (
-            f"{self.endpoint}/api/datasets/{repo_id}"
+            f"{self.endpoint}/api/spaces/{repo_id}"
             if revision is None
-            else f"{self.endpoint}/api/datasets/{repo_id}/revision/{revision}"
+            else f"{self.endpoint}/api/spaces/{repo_id}/revision/{revision}"
         )
         headers = {"authorization": f"Bearer {token}"} if token is not None else None
-        params = {"full": "true"}
-        r = requests.get(path, headers=headers, params=params, timeout=timeout)
```

Comment:
The `full: true` param is not needed anymore?

Reply:
it wasn't used for this endpoint on the server side